pkg/controller/egressgateway/egressgateway_controller.go

```diff
@@ -0,0 +1,666 @@
+		key := types.NamespacedName{Name: ipPool.Name}
+		err := cli.Get(ctx, key, instance)
+		if err != nil {
+			return err
+		}
+		if ipPool.CIDR != "" {
+			if instance.Spec.CIDR != ipPool.CIDR {
+				return fmt.Errorf("IPPool CIDR does not match with name")
+			}
+		}
+		if awsNativeIP == operatorv1.NativeIPEnabled && instance.Spec.AWSSubnetID == "" {
+			return fmt.Errorf("AWS subnet ID must be set when NativeIP is enabled")
+		}
+		return nil
+	}
+	if ipPool.CIDR != "" {
+		instance := &crdv1.IPPoolList{}
+		err := cli.List(ctx, instance)
+		if err != nil {
+			return err
+		}
+		for _, item := range instance.Items {
+			if item.Spec.CIDR == ipPool.CIDR {
+				if awsNativeIP == operatorv1.NativeIPEnabled && item.Spec.AWSSubnetID == "" {
+					return fmt.Errorf("AWS subnet ID must be set when NativeIP is enabled")
+				}
+				return nil
+			}
+		}
+	}
+	return fmt.Errorf("IPPool matching CIDR = %s not present", ipPool.CIDR)
+}
+
+func getUnreadyEgressGateway(egws []operatorv1.EgressGateway) *operatorv1.EgressGateway {
+	for _, egw := range egws {
+		if egw.Status.State != operatorv1.TigeraStatusReady {
+			return &egw
+		}
+	}
+	return nil
+}
+
+func setDegraded(cli client.Client, ctx context.Context, egw *operatorv1.EgressGateway, reason, msg string) {
+	updateEgwStatusConditions(cli, ctx, egw, operatorv1.ComponentDegraded, metav1.ConditionTrue, reason, msg, true)
+}
+
+func setProgressing(cli client.Client, ctx context.Context, egw *operatorv1.EgressGateway, reason, msg string) {
+	updateEgwStatusConditions(cli, ctx, egw, operatorv1.ComponentProgressing, metav1.ConditionTrue, reason, msg, false)
+}
+
+func setAvailable(cli client.Client, ctx context.Context, egw *operatorv1.EgressGateway, reason, msg string) {
+	updateEgwStatusConditions(cli, ctx, egw, operatorv1.ComponentAvailable, metav1.ConditionTrue, reason, msg, true)
+}
+
+func updateEgwStatusConditions(cli client.Client, ctx context.Context, egw *operatorv1.EgressGateway, ctype operatorv1.StatusConditionType, status metav1.ConditionStatus, reason, msg string, updateStatus bool) {
+	found := false
+	for idx, cond := range egw.Status.Conditions {
+		if cond.Type == string(ctype) {
+			cond.Status = status
+			cond.Reason = reason
+			cond.Message = msg
+			found = true
+		} else {
+			cond.Status = metav1.ConditionFalse
+			cond.Reason = string(operatorv1.Unknown)
+			cond.Message = ""
+		}
+		cond.LastTransitionTime = metav1.NewTime(time.Now())
+		egw.Status.Conditions[idx] = cond
+	}
+	if !found {
+		condition := metav1.Condition{Type: string(ctype), Status: status, Reason: reason, Message: msg, LastTransitionTime: metav1.NewTime(time.Now())}
+		egw.Status.Conditions = append(egw.Status.Conditions, condition)
+	}
+	if updateStatus {
+		if err := cli.Status().Update(ctx, egw); err != nil {
+			log.WithValues("Name", egw.Name, "Namespace", egw.Namespace, "error", err).Info("Error updating status")
+		}
+	}
+}
+
+func getDegradedMsg(egw *operatorv1.EgressGateway) string {
```

Comment:
We shouldn't need this function, IIUC. If we do the status handling properly in the status manager